reframe/frontend/executors/policies.py

```diff
@@ -27,6 +27,28 @@ def _cleanup_all(tasks, *args, **kwargs):
     tasks[:] = [t for t in tasks if t.ref_count]
 
 
+def format_duration(duration):
+    mssg = '[compile: '
+    if duration['compile_complete']:
+        mssg += f"{duration['compile_complete']:.3f} "
+    else:
+        mssg += 'n/a '
```

Comment:
I would avoid all these ifs here by defining a nested function to do this custom formating:

```python
def _tf(t):
    return f'{t:.3f}' if t else 'n/a'

msg = f"[compile: {_tf(duration['compile_complete'])}"
```